src/client/service-worker/events/utility.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  match, put, remove, getMetaData, setMetaData, createCacheName,
+} from '@americanexpress/one-service-worker';
+
+const localeRegExp = /(([a-z]{2,3})(?:-)?([a-zA-Z]{1,})?)\/[^/]*\.json/;
+const legacyRegExp = /legacy\.browser|\/legacy\//;
+
+export function createResourceMetaData(event, resourceInfo, revision) {
+  const [name, baseUrl] = resourceInfo;
+  const [version] = baseUrl.replace(/\/$/, '').split('/').reverse();
+  const { request } = event;
+
+  let type = name === 'app' ? 'one-app' : 'modules';
+
+  let path;
+  let locale;
+  if (localeRegExp.test(request.url)) {
+    type = 'lang-packs';
+    [path, locale] = request.url.match(localeRegExp);
+  }
+
+  let bundle = 'browser';
+  if (legacyRegExp.test(request.url)) {
+    bundle = 'legacy';
+  }
+
+  const metaData = {
+    bundle,
+    type,
+    name,
+    version,
+    path: path || request.url.replace(baseUrl, '').replace(/(.*)(\?[^/]*)$/, '$1'),
+    url: request.url,
+    cacheName: createCacheName(type),
+  };
+
+  if (revision) metaData.revision = revision;
+  if (locale) metaData.locale = locale;
+
+  return metaData;
+}
+
+export function markResourceForRemoval(cachedMetaRecord, newMetaRecord) {
+  if (cachedMetaRecord.revision !== newMetaRecord.revision) return true;
+  if (cachedMetaRecord.version !== newMetaRecord.version) return true;
+  if (cachedMetaRecord.bundle !== newMetaRecord.bundle) return true;
```

Comment:
is this needed ? the url will differ between a legacy and modern module so they should never have a cache collision

Reply:
It isn't needed per se, however it does invalidate if the "same" version comes in, or a browser upgrades for example. It would then remove the previous bundle variant.

Reply:
If we don't care to do this, I can remove

Reply:
from my understanding `bundle` is either `legacy` or` browser`, if it does not change then this will return false so it will not invalidate. also not sure how a browser upgrading will alter the result of this condition ?

Reply:
Yes exactly, invalidation would remain false - it would only invalidate when the `browserlist` used to generate `last 2 versions` for modern bundles is updated or a user lags behind a few version updates and decides to update their browser, or if a user has to downgrade for whatever reason/restriction.